includes/discovery/sensors/current/sentry3.inc.php

```diff
@@ -1,105 +0,0 @@
-<?php
-
-$divisor = '100';
-$outlet_divisor = $divisor;
-$multiplier = '1';
-
-// These PDUs may have > 1 "tower" accessible via a single management interface
-$tower_count = snmp_get($device, 'systemTowerCount.0', '-Ovq', 'Sentry3-MIB');
-$towers = 1;
-while ($towers <= $tower_count) {
-    // Check for Infeeds
-    $infeed_oids = snmp_walk($device, "infeedID.$towers", '-Osqn', 'Sentry3-MIB');
-    d_echo($infeed_oids . "\n");
-
-    $infeed_oids = trim($infeed_oids);
-
-    if ($infeed_oids) {
-        echo 'ServerTech Sentry Infeed ';
-    }
-
-    foreach (explode("\n", $infeed_oids) as $infeed_data) {
-        $infeed_data = trim($infeed_data);
-        if ($infeed_data) {
-            [$infeed_oid,$descr] = explode(' ', $infeed_data, 2);
-            $split_oid = explode('.', $infeed_oid);
-            $infeed_index = $split_oid[count($split_oid) - 1];
-
-            // infeedLoadValue
-            $infeed_oid = '.1.3.6.1.4.1.1718.3.2.2.1.7.' . $towers . '.' . $infeed_index;
-
-            $descr_string = snmp_get($device, "infeedID.$towers.$infeed_index", '-Ovq', 'Sentry3-MIB');
-            $descr = "Infeed $descr_string";
```

Comment:
current description is built as "Infeed " followed by the infeedID. 
YAML equivalent would be:
`descr: 'Infeed BA {{ $infeedID }}`

Reply:
Making use of the `infeedName`s, rather than just the IDs, was one of the motivations for this.

I see that the test snmprec doesn't have this - is there coalesce functionality along the lines of `descr: '{{ $infeedName ?? Infeed {{ $infeedID }} }}'` that would satisfy the test (and devices with it missing)?

https://github.com/librenms/librenms/blob/dd7e03fffc15dc0394e9909c901ddbf7d95ef92f/tests/snmpsim/sentry3.snmprec#L297-L299

![image](https://github.com/librenms/librenms/assets/1735973/e4e182ca-0eac-47ef-adcd-884f1fbc6414)
```
Sentry3-MIB::infeedID.1.1 = STRING: AA
Sentry3-MIB::infeedID.1.2 = STRING: AB
Sentry3-MIB::infeedID.1.3 = STRING: AC
Sentry3-MIB::infeedID.1.4 = STRING: AD
Sentry3-MIB::infeedName.1.1 = STRING: TowerA_InfeedA
Sentry3-MIB::infeedName.1.2 = STRING: TowerA_InfeedB
Sentry3-MIB::infeedName.1.3 = STRING: TowerA_InfeedC
Sentry3-MIB::infeedName.1.4 = STRING: TowerA_Neutral
```

Reply:
Then you'll have to re-generate the tests for the description part. This is clearly not a problem. As long as the index is kept, you can change the description if the new one mayes more sense (and it seems to make more sense as far as I see :) )